okhttp/src/test/java/okhttp3/HttpUrlTest.java

```diff
@@ -17,409 +17,466 @@
 
 import java.net.URI;
 import java.net.URL;
-import java.util.Collection;
 import java.util.Collections;
 import okhttp3.UrlComponentEncodingTester.Component;
 import okhttp3.UrlComponentEncodingTester.Encoding;
-import org.junit.Ignore;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
 
-import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.fail;
 
-@RunWith(Parameterized.class)
 public final class HttpUrlTest {
```

Comment:
For tests like this should we just do subclasses? That's probably a lot less invasive in the syntax.